superbench/benchmarks/micro_benchmarks/disk_performance.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the Disk Performance benchmarks."""
+
+import json
+import os
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class DiskPerformance(MicroBenchmarkWithInvoke):
+    """The KernelLaunch overhead benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'fio'
+
+        self._io_patterns = ['seq', 'rand']
+        self._io_types = ['read', 'write']
+        self._rand_block_size = 4 * 1024    # 4KiB
+        self._seq_block_size = 128 * 1024    # 128KiB
+        self._default_iodepth = 64
+        self._default_ramp_time = 10
+        self._default_runtime = 60
+        self._default_numjobs_for_rand = 4
+
+        self._common_fio_args =\
+            ' --randrepeat=1 --thread=1 --ioengine=libaio --direct=1'\
+            ' --norandommap=1 --lat_percentiles=1 --group_reporting=1'\
+            ' --output-format=json'
+        self._fio_args = {}
+
+        # Sequentially write 128KiB to the device twice
+        self._fio_args['seq_precond'] = self._common_fio_args +\
+            ' --name=seq_precond --rw=write --bs=%d --iodepth=%d --numjobs=1 --loops=2' %\
+            (self._seq_block_size, self._default_iodepth)
+
+        # Randomly write 4KiB to the device
+        self._fio_args['rand_precond'] = self._common_fio_args +\
+            ' --name=rand_precond --rw=randwrite --bs=%d --iodepth=%d --numjobs=%d --time_based=1' %\
+            (self._rand_block_size, self._default_iodepth, self._default_numjobs_for_rand)
+
+        # Seq/rand read/write tests
+        for io_pattern in self._io_patterns:
+            for io_type in self._io_types:
+                io_str = '%s_%s' % (io_pattern, io_type)
+                fio_rw = io_type if io_pattern == 'seq' else io_pattern + io_type
+                fio_bs = self._seq_block_size if io_pattern == 'seq' else self._rand_block_size
+                self._fio_args[io_str] = self._common_fio_args +\
+                    ' --name=%s --rw=%s --bs=%d --time_based=1' % (io_str, fio_rw, fio_bs)
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--filenames',
```

Comment:
rename to `--block_devices`? what if a regular file path is provided?

Reply:
Fixed